Adapt smart proxy upgrade for katello (#518)
theforeman/foreman-documentation: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc

+
For information on backups, see {AdministeringDocURL}backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProjectVersionPrevious}_ guide.

ifdef::katello[]
+

> . Regenerate certificates.  On the main Foreman/Katello server:

Reviewer: Please convert to one sentence per line & use asciidoc attributes instead of `Foreman`: https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L80

Reviewer 2: @maximiliankolb this guide is new and was brought upstream quite fast. I think there are probably problems with one sentence per line rule violations throughout. If it is the case, I can spend a while later today fixing this rather than blocking this PR (It's kind of my fault for not doing this upfront but this is the last guide to come upstream).

Would you be OK with that?

[options="nowrap" subs="attributes"]
----

> # yum update -y http://yum.theforeman.org/katello/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm \

Reviewer 3: ```suggestion
# yum update -y https://yum.theforeman.org/katello/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm \
```

                       --certs-update-all \
                       --certs-tar    "~/myproxy.example.com-certs.tar"
----
+

> . Copy the resulting tarball to your Smart Proxy, for this example we will use /root/myproxy.example.com-certs.tar

Reviewer: ```suggestion
. Copy the resulting tarball to your {SmartProxy}, for this example we will use `/root/myproxy.example.com-certs.tar`.
```

----

> # yum update -y http://yum.theforeman.org/katello/{KatelloVersion}/katello/el8/x86_64/katello-repos-latest.rpm \

Reviewer 3: ```suggestion
# yum update -y https://yum.theforeman.org/katello/{KatelloVersion}/katello/el8/x86_64/katello-repos-latest.rpm \
```

                https://yum.theforeman.org/releases/{ProjectVersion}/el8/x86_64/foreman-release.rpm
----
. Clean yum cache:
+
----

> # yum clean all

Reviewer 3: Wouldn't it be sufficient to use `yum clean metadata`?

Reviewer 3: You now only updated Satellite, not the upstream version.

Reviewer 2: Does this matter to you @ekohl because your comment still stands. Everything else looks good to me.

----
+
. Update Packages:
+
----
# yum update -y
----
+
. Regenerate certificates.  On the main Foreman/Katello server:
+
----
# foreman-proxy-certs-generate --foreman-proxy-fqdn "myproxy.example.com" \
                       --certs-update-all \
                       --certs-tar    "~/myproxy.example.com-certs.tar"
----
+

> . Copy the resulting tarball to your Smart Proxy

Reviewer 3: I think I would move this part up to line 39.

+

> . Run the installers:

Reviewer 3: ```suggestion
. Run the installer:
```

Author: thx, fixed!

+
For information on backups, see {AdministeringDocURL}backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProjectVersionPrevious}_ guide.

ifdef::katello[]

> . Update repositories for el7

Reviewer 3: Thought for the future: generate EL7 / EL8 manuals instead of just EL. Also, I'd capitalize it.
```suggestion
. Update repositories for EL7
```

+
[options="nowrap" subs="attributes"]
----

> # yum update -y http://yum.theforeman.org/katello/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm
> # yum update -y https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm

Reviewer 3: While not today needed, I think this needs to be a single command.
```suggestion
# yum update -y http://yum.theforeman.org/katello/{KatelloVersion}/katello/el7/x86_64/katello-repos-latest.rpm https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm
```

The reason is that `katello-repos` requires `foreman-release`. That's why I think you should update `foreman-release` first.

I also think that we should actually update the spec to also require specific releases so there can't be a version mismatch. We did this with katello itself, but if you get it earlier, it's even better.

That means updating this line:
https://github.com/theforeman/foreman-packaging/blob/f2421fa4d7cbeabf8f7974a02525c20fbd7e9c41/packages/katello/katello-repos/katello-repos.spec#L24

If we already have the documentation ready for that, we can apply it any time we want.

(same below for EL8)

Author: sounds good to me

----

> . Update repositories for el8

Reviewer 3: ```suggestion
. Update repositories for EL8
```

+
[options="nowrap" subs="attributes"]
----
# yum update -y http://yum.theforeman.org/katello/{KatelloVersion}/katello/el8/x86_64/katello-repos-latest.rpm
# yum update -y https://yum.theforeman.org/releases/{ProjectVersion}/el8/x86_64/foreman-release.rpm
----
. Clean yum cache:
+
----
# yum clean all
----
+
. Update Packages:
+
----
# yum update -y
----
+
. Regenerate certificates.  On the main Foreman/Katello server:
+
----

> # foreman-proxy-certs-generate --foreman-proxy-fqdn "myproxy.example.com"\

Reviewer 3: This may be me, but I prefer a space before the backslash to clearly separate it.
```suggestion
# foreman-proxy-certs-generate --foreman-proxy-fqdn "myproxy.example.com" \
```

I'd also keep the downtime window smaller and do this first, then upgrade the actual content proxy. Then you have everything in place.

                       --certs-update-all\
                       --certs-tar    "~/myproxy.example.com-certs.tar"
----
+
. Copy the resulting tarball to your Smart Proxy
+

> +

Reviewer: This whitespace between bullet points in not necessary.
